solveMuscleRedundancy.m

```diff
@@ -26,126 +25,117 @@
 end
 
-% get indexes of the muscles for which optimal fiber length, tendon stiffness are estimated
-[DatStore] = GetIndices_OptMuscleParams(DatStore,Misc);
-
-% set the default value of the tendon stiffness
-if isfield(Misc,'Set_kT_ByName') && ~isempty(Misc.Set_kT_ByName)
-    [Misc,DatStore] = set_kT_ByName(Misc,DatStore);
+%% From here only use the trails selected
+% ----------------------------------------------------------------------- %
+% Extract muscle information -------------------------------------------- %
+% Get number of degrees of freedom (dofs), muscle-tendon lengths, moment
+% arms, stiffness and shift for the selected muscles.
+for trial = Misc.trials_sel
+    [~,Misc.MAtrialName{trial},~]=fileparts(Misc.IKfile{trial});
 end
 
-% Structure EMG data & select corresponding muscles in model
-[DatStore] = GetEMGInfo(Misc,DatStore);
+Misc = getMuscles4DOFS(Misc);
+
+[Misc,DatStore] = getMuscleInfo(Misc,DatStore);
+    
+% display warnings in muscle selection
+[Misc] = Warnings_MuscleNames(DatStore,Misc);
+
+% get indexes of the muscles for which optimal fiber length, tendon stiffness are estimated
+[DatStore,Misc] = GetIndices(DatStore,Misc);
 
-% Structure ultrasound data & select corresponding muscles in model
-[DatStore] = GetUSInfo(Misc,DatStore);
+% get the EMG information
+[Misc,DatStore] = GetEMGInfo(Misc,DatStore);
+% [DatStore] = GetEMGInfo_DG(Misc,DatStore);
+[Misc,DatStore] = GetUSInfo(Misc,DatStore);
 
 % get the number of muscles
-NMuscles = length(DatStore(1).MuscleNames);
+for trial = Misc.trials_sel
+    NMuscles(trial) = DatStore(trial).nMuscles;
+end
 
 %% Static optimization
 % ----------------------------------------------------------------------- %
-
 % Solve the muscle redundancy problem using static optimization
 % NOTE: We do not estimate any parameters here, but these results can serve as
 % decent initial guess for the later dynamic optimization
-
 % Extract the muscle-tendon properties
-[Misc.params,Misc.lMo,Misc.lTs,Misc.FMo,Misc.alphao]=ReadMuscleParameters(model_path,DatStore(1).MuscleNames);
-
-% Static optimization using IPOPT solver (used as an initial guess)
-for trial = 1:Misc.nTrials
-    DatStore = SolveStaticOptimization_IPOPT_CasADi(DatStore,Misc,trial);
+for i=Misc.trials_sel
+    % Static optimization using IPOPT solver (used as an initial guess)
+    DatStore = SolveStaticOptimization_IPOPT_CasADi(DatStore,Misc,i);
 end
 
 %% Input activation and contraction dynamics
 % ----------------------------------------------------------------------- %
-tau_act = 0.015;    Misc.tauAct = tau_act * ones(NMuscles, 1);     % activation time constant (activation dynamics)
-tau_deact = 0.06;   Misc.tauDeact = tau_deact * ones(NMuscles,1);  % deactivation time constant (activation dynamics)
+Misc.tau_act = 0.015;    
+Misc.tau_deact = 0.06;   
 Misc.b = 0.1;       % tanh coefficient for smooth activation dynamics
 
-Misc.kT=Misc.kT;
-Misc.shift=Misc.shift;
+% Misc.kT=Misc.kT;
```

Comment:
this should be removed and not in comment

Reply:
done